src/ICSharpCode.SharpZipLib/GZip/GZip.cs

```diff
@@ -18,7 +18,7 @@ public static class GZip
 		/// <param name="outStream">The output stream to receive the decompressed data.</param>
 		/// <param name="isStreamOwner">Both streams are closed on completion if true.</param>
 		/// <exception cref="ArgumentNullException">Input or output stream is null</exception>
-		public static void Decompress(Stream inStream, Stream outStream, bool isStreamOwner)
+		public static void Decompress(Stream? inStream, Stream? outStream, bool isStreamOwner)
```

Comment:
should these streams be nullable if the function throws on null?

Reply:
Not sure, I don't think we can check for null if we say that they're non-nullable. I have to read up on how to interface with consumer code with and without Nullable enabled.

Reply:
Callers without Nullable enabled will just behave as they do before (i.e treat it as though it can be null), and you can still check for null and throw for those cases (or in general really, given that the checks don't actually *prevent* it from being null even if marked as non-nullable).

FxCop still gives warnings that non-nullable parameters to public functions should be checked before use as well.

Reply:
Yep you're right. There are some instances where you branch on null that the compiler thinks it should be nullable.